### PR TITLE
Add rsync to build image

### PR DESCRIPTION
**Pull request by radar**
## Approvals

- [ ] Verified
- [ ] @hebbo
- [ ] @ptolemy428
- [ ] @iiman

### CODE REVIEW

**Comment by radar** [comment]
This is part one of two PRs which will speed up the cloudformation-stacks script. You're using `copy` in your Ansible script, which is actually quite slow. It takes a few minutes to run. The `synchronize` command is _much_ faster and should be used instead, but it needs `rsync` installed in the image before you can run it.

Here's a PR which adds `rsync` to the image.

**Comment by frenchi** [comment]
Hmm interesting @radar. LGTM 👍 

/nerd-snipe
Is that because of `rsync`'s compression? I wonder how it compares to
`scp -C -o 'CompressionLevel 9'`

or is it because it only copies diffs? In which case we should expect comparable performance on clean boxes.

Either way, if it makes a big time difference 👍

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,9 @@
 FROM cultureamp/docker-python:latest
 MAINTAINER Infrastructure Services Team <is_team@cultureamp.com>
 
+RUN apt-get update
```
Comment (frenchi) [inline Dockerfile]: docker creates intermediate images for each command run, so you could probably squish these in to:
`RUN apt-get update && apt-get install rsync`

**Comment by radar** [comment]
> Is that because of rsync's compression? 

I don't know. 

What I _do_ know is that if you're copying a large batch of files with Ansible, you should use `synchronize`. I read it on the internet :P It's worked for me in the past, and looks like it works here too. The `copy` command appears to copy each file one at a time:

```
<127.0.0.1> EXEC /bin/sh -c 'LANG=C.UTF-8 LC_ALL=C.UTF-8 LC_MESSAGES=C.UTF-8 /usr/local/bin/python2 /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/copy && sleep 0'
<127.0.0.1> PUT /tmp/tmpkC_kLc TO /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/stat
<127.0.0.1> EXEC /bin/sh -c 'LANG=C.UTF-8 LC_ALL=C.UTF-8 LC_MESSAGES=C.UTF-8 /usr/local/bin/python2 /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/stat && sleep 0'
<127.0.0.1> PUT /usr/src/app/roles/cfn/files/account-bootstrapping-common-resources/account-bootstrapping-common-resources_main.py TO /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/source
<127.0.0.1> PUT /tmp/tmpMfNwtj TO /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/copy
<127.0.0.1> EXEC /bin/sh -c 'LANG=C.UTF-8 LC_ALL=C.UTF-8 LC_MESSAGES=C.UTF-8 /usr/local/bin/python2 /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/copy && sleep 0'
<127.0.0.1> PUT /tmp/tmpoeAQL7 TO /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/stat
<127.0.0.1> EXEC /bin/sh -c 'LANG=C.UTF-8 LC_ALL=C.UTF-8 LC_MESSAGES=C.UTF-8 /usr/local/bin/python2 /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/stat && sleep 0'
<127.0.0.1> PUT /usr/src/app/roles/cfn/files/account-bootstrapping-common-resources/deployment_iam_role.py TO /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/source
<127.0.0.1> PUT /tmp/tmpn238um TO /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/copy
<127.0.0.1> EXEC /bin/sh -c 'LANG=C.UTF-8 LC_ALL=C.UTF-8 LC_MESSAGES=C.UTF-8 /usr/local/bin/python2 /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/copy && sleep 0'
<127.0.0.1> PUT /tmp/tmplfH2FE TO /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/stat
<127.0.0.1> EXEC /bin/sh -c 'LANG=C.UTF-8 LC_ALL=C.UTF-8 LC_MESSAGES=C.UTF-8 /usr/local/bin/python2 /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/stat && sleep 0'
<127.0.0.1> PUT /usr/src/app/roles/cfn/files/account-bootstrapping-common-resources/config/root.yml TO /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/source
<127.0.0.1> PUT /tmp/tmpbZbDAL TO /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/copy
<127.0.0.1> EXEC /bin/sh -c 'LANG=C.UTF-8 LC_ALL=C.UTF-8 LC_MESSAGES=C.UTF-8 /usr/local/bin/python2 /root/.ansible/tmp/ansible-tmp-1477950378.5-189228046574959/copy && sleep 0'
```

I don't know the technical reasons as to why it's slow, sorry.

**Comment by radar** [comment]
I've merged the two `apt-get` commands together. Thanks!

**Comment by iiman** [comment]
@radar You have not pushed your changes

**Comment by radar** [comment]
You're right, I hadn't! Try now :)

**Comment by vgunawan** [comment]
AFAIK, rsync only copies the diff, it's a smarter scp. It's doing it by invoking rsync program on the remote(target) host. Also it runs multi threaded.

**Pull request merged**